### PR TITLE
Update schema object limits

### PR DESCRIPTION
**Pull request by rmloveland**
Fixes DOC-10457

Summary of changes:

- Update 'Schema Design Overview' with new higher values for # of tables in single- and multi-region clusters

- Also note that Backup/CDC will also affect scale/performance

### CODE REVIEW

**Comment by rmloveland** [comment]
@fqazi and @dikshant please take a look and let me know if:

- anything is missing or wrong
- should we backport this docs change to v24.1+?  or is it v24.3 only?

**Review by fqazi — APPROVED** [review]

**Review by dikshant — CHANGES_REQUESTED** [review]
@rmloveland I was discussing this with @kevin-v-ngo and our feeling is that starting with 24.3 DB console will work okay with 10k tables. But 25k might be a stretch. I think we should just say 10k here. A single number also creates less confusion.

File: src/current/v24.3/schema-design-overview.md [hunk 1]
```diff
@@ -120,12 +120,18 @@ The following table lists specific limits imposed by CockroachDB.
...
+- 25,000 tables for single region clusters.
+- 10,000 tables for [multi-region clusters]({% link {{ page.version.version }}/multiregion-overview.md %}).
+
+Greater numbers are possible, depending on the complexity of the tables (number of columns and indexes) and hardware specifications.
 
 As you scale to a large number of tables, note that:
 
 - The amount of RAM per node is the limiting factor for the number of tables and other schema objects the cluster can support. This includes columns, indexes, GIN indexes, constraints, and partitions. Increasing RAM is likely to have the greatest impact on the number of these objects that a cluster can support, while increasing the number of nodes will not have a substantial effect.
 - The number of databases or schemas on the cluster has minimal impact on the total number of tables that it can support.
+- Performance at larger numbers of tables may be affected by your use of [Backups]({% link {{ page.version.version }}/backup-and-restore-overview.md %}) and [Change Data Capture]({% link {{ page.version.version }}/change-data-capture-overview.md %}).
```
Comment (dikshant) [inline src/current/v24.3/schema-design-overview.md:134]: IIRC it was restore that would be affected more so than backups. @alicia-l2 can confirm!

**Review by dikshant — APPROVED** [review]
LGTM!

File: src/current/v24.3/schema-design-overview.md (hunk 1 above)
Comment (alicia-l2) [reply src/current/v24.3/schema-design-overview.md:134]: Yes maybe we could just say Backup/Restore

**Comment by rmloveland** [comment]
> @rmloveland I was discussing this with @kevin-v-ngo and our feeling is that starting with 24.3 DB console will work okay with 10k tables. But 25k might be a stretch. I think we should just say 10k here. A single number also creates less confusion.

sounds good, updated to just say 10k

File: src/current/v24.3/schema-design-overview.md (hunk 1 above)
Comment (rmloveland) [reply src/current/v24.3/schema-design-overview.md]: for now i will update to say "backup and restore" to keep it vague.  Alicia, happy to update next week with your feedback, rushing to get this in for Monday's release at the moment :-)

File: src/current/v24.3/schema-design-overview.md [hunk 2]
```diff
@@ -120,12 +120,15 @@ The following table lists specific limits imposed by CockroachDB.
...
+CockroachDB has been shown to perform well with clusters containing 10,000 tables. Greater numbers are possible, depending on the complexity of the tables (number of columns and indexes) and hardware specifications.
+
+Greater numbers are possible, depending on the complexity of the tables (number of columns and indexes) and hardware specifications.
 
 As you scale to a large number of tables, note that:
 
 - The amount of RAM per node is the limiting factor for the number of tables and other schema objects the cluster can support. This includes columns, indexes, GIN indexes, constraints, and partitions. Increasing RAM is likely to have the greatest impact on the number of these objects that a cluster can support, while increasing the number of nodes will not have a substantial effect.
 - The number of databases or schemas on the cluster has minimal impact on the total number of tables that it can support.
+- Performance at larger numbers of tables may be affected by your use of [backup and restore]({% link {{ page.version.version }}/backup-and-restore-overview.md %}) and [Change Data Capture]({% link {{ page.version.version }}/change-data-capture-overview.md %}).
```
Comment (taroface) [inline src/current/v24.3/schema-design-overview.md:131]:
```suggestion
- Performance at larger numbers of tables may be affected by your use of [backup and restore]({% link {{ page.version.version }}/backup-and-restore-overview.md %}) and [Change data capture (CDC)]({% link {{ page.version.version }}/change-data-capture-overview.md %}).
```

**Review by taroface — APPROVED** [review]
LGTM, formatting nit

File: src/current/v24.3/schema-design-overview.md (hunk 1 above)
Comment (rmloveland) [reply src/current/v24.3/schema-design-overview.md]: omg @alicia-l2 i totally missed your comment but am seeing it now, sorry about that - maybe i needed to refresh the page

luckily i wrote "backup and restore" :-)

**Pull request merged**